crates/sources/src/loader.rs

```diff
@@ -125,18 +134,45 @@ impl<F: Fetcher> Loader<F> {
                     .binary_search_by_key(&resource_url.as_str(), |r| r.resource.as_str())
                     .ok()
                     .and_then(|ind| resources.get(ind))
-                    .and_then(|r| serde_yaml::from_slice(&r.content).ok())
-                    .map(|config| {
-                        serde_json::to_value(models::ConnectorConfig {
-                            image,
-                            config: models::Config::Inline(config),
-                        })
-                        .unwrap()
+                    .and_then(|r| {
+                        // transcode the YAML into a JSON RawValue to avoid re-ordering of elements
+                        // in the intermediary Value (which is a BTreeMap) since this could be a sops encrypted
+                        // configuration and the ordering of the elements must be preserved. See #303.
+                        let mut buf = Vec::new();
+                        let writer = BufWriter::new(&mut buf);
+                        let deserializer = serde_yaml::Deserializer::from_slice(&r.content);
+                        tracing::debug!("r.content {:?}", r.content);
+                        let mut serializer = serde_json::Serializer::new(writer);
+
+                        let transcode_result =
+                            serde_transcode::transcode(deserializer, &mut serializer)
+                                .map_err(|e| LoadError::YAMLTranscodeErr(e));
+
+                        serializer.into_inner().flush().unwrap();
+                        let raw_value_result =
+                            RawValue::from_string(String::from_utf8(buf).unwrap())
+                                .map_err(|e| LoadError::JSONErr(e));
+
+                        // TODO: handle these errors properly (similar to self.fallible)
```

Comment:
I couldn't call `self.fallible` here because we expand it at the beginning of the function, and so we have partially moved it, so it can no longer be used as a whole.

One thing we could do is create a function similar to `self.fallible`, that doesn't rely on `self` as a whole, but accepts the `errors` table as input and operates on that, then we can have `errors` as part of our expansion and use that function I think. WDYT?